pkgs/by-name/da/databricks-cli/package.nix

```diff
@@ -0,0 +1,53 @@
+{ lib
+, stdenv
+, fetchurl
+, unzip
+}:
+
+stdenv.mkDerivation rec {
+  pname = "databricks-cli";
+  version = "0.209.1";
+
+  src = {
```

Comment:
What about building from sources?

Reply:
I suppose I could - I guess I am not sure what the benefit would be, in this case? The CLI is [packaged as a dependency free executable](https://github.com/databricks/cli/blob/main/README.md?plain=1#L13), so I don't see any issue with building from the releases they offer. If there are particular benefits or practices in nixpkgs that favor building from source, I'd be interested to hear about them.

I'm open to the idea of changing the package if it makes sense. Your advice would be helpful here! Especially since I'm still getting the hang of contributing.

Reply:
> I'm open to the idea of changing the package if it makes sense. Your advice would be helpful here! Especially since I'm still getting the hang of contributing.

Here you are:
```nix
{ lib
, stdenv
, fetchFromGitHub
, buildGoModule
, testers
, databricks-cli
}:

buildGoModule rec {
  pname = "databricks-cli";
  version = "0.209.1";

  src = fetchFromGitHub {
    owner = "databricks";
    repo = "cli";
    rev = "v${version}";
    hash = "sha256-6qsVMb5YVERqlJ2dLqTqmqzO6x5mjqe1dCm6eX9tt5U=";
  };

  vendorHash = "sha256-Urnwlj/J08AMz4M09wVqXvL3wzr3646IZDvu9XuT+yY=";

  postPatch = ''
    rm -fr bundle/internal/tf/codegen
  '';

  ldflags = [
    "-s"
    "-w"
    "-X github.com/databricks/cli/internal/build.buildVersion=${version}"
  ];

  postInstall = ''
    mv $out/bin/{cli,databricks-cli}
  '';

  doCheck = false;

  passthru.tests.version = testers.testVersion {
    package = databricks-cli;
    version = "v${version}";
  };

  meta = with lib; {
    description = "A command-line interface for Databricks";
    homepage = "https://github.com/databricks/cli";
    license = licenses.asl20;
    platforms = platforms.unix;
    maintainers = with maintainers; [ jl178 ];
    mainProgram = "databricks-cli";
  };
}
```